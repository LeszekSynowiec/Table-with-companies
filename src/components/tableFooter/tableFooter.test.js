import React from "react";
import TableFooter from "./tableFooter.component";
import { shallow } from "enzyme";

describe("TableFooter component", () => {
  it("shoud render with out crashing", () => {
    const currentPage = 1;
    const handleSelectPage = jest.fn();
    const handleSearchString = jest.fn();
    const numberOfPages = 30;
    const component = shallow(
      <TableFooter
        currentPage={currentPage}
        handleSelectPage={handleSelectPage}
        handleSearchString={handleSearchString}
        numberOfPages={numberOfPages}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it(' handleSelectPage shoud be called', ()=> {
    const currentPage = 1;
    const handleSelectPage = jest.fn();
    const handleSearchString = jest.fn();
    const numberOfPages = 30;
    const component = shallow(
      <TableFooter
        currentPage={currentPage}
        handleSelectPage={handleSelectPage}
        handleSearchString={handleSearchString}
        numberOfPages={numberOfPages}
      />);
        component.find('select').simulate('click')
        expect(handleSelectPage).toHaveBeenCalled()
  })

  it(' handleSearchString shoud be called', ()=> {
    const currentPage = 1;
    const handleSelectPage = jest.fn();
    const handleSearchString = jest.fn();
    const numberOfPages = 30;
    const component = shallow(
      <TableFooter
        currentPage={currentPage}
        handleSelectPage={handleSelectPage}
        handleSearchString={handleSearchString}
        numberOfPages={numberOfPages}
      />);
        component.find('input').simulate('change')
        expect(handleSearchString).toHaveBeenCalled()
  })
});
