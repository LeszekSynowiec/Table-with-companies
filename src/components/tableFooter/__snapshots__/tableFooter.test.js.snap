// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFooter component shoud render with out crashing 1`] = `ShallowWrapper {}`;
